### PR TITLE
Add validation for saveWebClientState, postMovie, and postScreenshot

### PR DESCRIPTION
**Pull request by dgarciabriseno**
This change adds a json schema validator to strictly validate the contents of the web client state being saved, and validate the given arguments for postMovie and postScreenshot.

This is implemented via a new "schema" validator on the existing InputValidator.
JSON schemas should be placed in the `management/schema` and then copied to `docroot/schema` via `management/schema/deploy.sh`.
- Schema files in the `management/schema` may contain template parameters populated by the database. 
- Schema files in `docroot/schema` are final schema files that may be accessed globally by any json schema validator (not only helioviewer).

To validate a json field, the input to the validator looks like this:
```
$expected = array(
  'schema' => array('json' => 'https://api.helioviewer.org/schema/some_schema.json')
)
```

Where the first `schema` key tells the validator to use the schema validator, and the second array is in the form `input_parameter_key => schema_file`.

### CODE REVIEW

File: management/schema/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+DOCROOT_DIR=../../docroot
```
Comment (mudhoney) [inline management/schema/deploy.sh:1]: We might need 
`#!/bin/bash`
maybe `set -e`

File: management/schema/image_layer.schema.template.json [hunk 2]
```diff
@@ -0,0 +1,87 @@
+{
```
Comment (dgarciabriseno) [inline management/schema/image_layer.schema.template.json:1]: Move other schemas to docroot, leave template file here.

File: management/schema/validate.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from argparse import ArgumentParser, Namespace
+import json
+from jsonschema import validate
+
+def parse_args() -> Namespace:
+    parser = ArgumentParser(description="Validates a json file against a json schema")
+    parser.add_argument("schema", type=str, help="JSON Schema file to validate against")
+    parser.add_argument("json", type=str, help="JSON file to validate")
+    return parser.parse_args()
```
Comment (dgarciabriseno) [inline management/schema/validate.py:9]: See if I can set a "prefix" like in the php validator

File: tests/unit_tests/validation/SchemaValidatorTest.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        foreach ($invalid_files as $file) {
+            if ($file == "." || $file == "..") {
+                continue;
+            }
+            $test_data["invalid/".$file] = SHOULD_FAIL;
+        }
+        return $test_data;
+    }
+
+    public function test_ValidateJsonSchema(): void
```
Comment (dgarciabriseno) [inline tests/unit_tests/validation/SchemaValidatorTest.php:52]: Use phpunit dataprovider

File: tests/unit_tests/module/WebClientTest.php [hunk 5]
```diff
@@ -45,4 +45,47 @@ public function test_getTile_BadDiffTime(): void
...
+          throw new Exception("Failed to read test json file.");
+        }
+        $json = json_decode($json_string, true);
+        $params = array(
+          'action' => 'saveWebClientState',
+          'json' => $json
+        );
+        // Set up the client
+        $client = new Module_WebClient($params);
+        $this->expectException(InvalidArgumentException::class);
```
Comment (dgarciabriseno) [inline tests/unit_tests/module/WebClientTest.php:87]: Could do expect exception message

**Pull request merged**